btree.go

```diff
@@ -64,6 +64,51 @@ type Item interface {
 	Less(than Item) bool
 }
 
+// FreeList represents a free list of btree nodes. By default each
+// BTree has its own FreeList, but multiple BTrees can share the same
+// FreeList.
+type FreeList struct {
+	freelist []*node
+	grow     bool
+}
+
+func NewFreeList() *FreeList {
+	return newFreeList(true)
+}
+
+func newDefaultFreeList() *FreeList {
+	return newFreeList(false)
+}
+
+func newFreeList(grow bool) *FreeList {
```

Comment:
Cool, let me know the results of your experiment.  If bounded works for
now, maybe we'll stick with that.  We can always add on later.

On Thu, Apr 7, 2016 at 2:09 PM, keep94 <notifications@github.com> wrote:

> In btree.go <https://github.com/google/btree/pull/4#discussion_r58936256>:
>
> > +// BTree has its own FreeList, but multiple BTrees can share the same
> > +// FreeList.
> > +type FreeList struct {
> > +	freelist []*node
> > +	grow     bool
> > +}
> > +
> > +func NewFreeList() *FreeList {
> > +	return newFreeList(true)
> > +}
> > +
> > +func newDefaultFreeList() *FreeList {
> > +	return newFreeList(false)
> > +}
> > +
> > +func newFreeList(grow bool) *FreeList {
>
> There is something to be said about simple API. I am willing to run
> experiments to see how big my freelist actually gets to see if it stays <
> 32 list. If it stays small enough, I'd be willing to keep the original
> bounded freelist.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/google/btree/pull/4/files/6736c10c68569737fa4c4a6ad3e71484808ef1c6#r58936256>
>